src/daft-io/src/s3_like.rs

```diff
@@ -49,6 +55,44 @@ pub(crate) struct S3LikeSource {
     anonymous: bool,
 }
 
+#[derive(Debug)]
```

Comment:
would be cleaner to have this in a separate file and only load the module if python feature is enabled.